src/physics.F90

```diff
@@ -1359,20 +1324,20 @@ subroutine inelastic_scatter(nuc, rxn, p)
       E_cm = E
 
       ! determine outgoing energy in lab
+      A = nuc%awr
       E = E_cm + (E_in + TWO * mu * (A+ONE) * sqrt(E_in * E_cm)) &
            / ((A+ONE)*(A+ONE))
 
       ! determine outgoing angle in lab
       mu = mu * sqrt(E_cm/E) + ONE/(A+ONE) * sqrt(E_in/E)
-
-      ! Because of floating-point roundoff, it may be possible for mu to be
-      ! outside of the range [-1,1). In these cases, we just set mu to exactly
-      ! -1 or 1
-
-      if (abs(mu) > ONE) mu = sign(ONE,mu)
     end if
 
-    ! Set outgoing energy and scattering angle
+    ! Because of floating-point roundoff, it may be possible for mu to be
+    ! outside of the range [-1,1). In these cases, we just set mu to exactly -1
+    ! or 1
+    if (abs(mu) > ONE) mu = sign(ONE,mu)
```

Comment:
The same check appears for angle distributions (which appear in uncorrelated angle-energy distributions). The reason there is another check here is because the LAB to COM transformation (if applied) changes the value of mu.